src/backend/joanie/core/serializers.py

```diff
@@ -396,24 +396,24 @@ class EnrollmentSerializer(serializers.ModelSerializer):
 
     id = serializers.CharField(read_only=True, required=False)
     course_run = CourseRunSerializer(read_only=True)
+    has_been_created_with_order = serializers.BooleanField(read_only=True)
 
     class Meta:
         model = models.Enrollment
         fields = ["id", "course_run", "is_active", "state"]
-        read_only_fields = ["id", "state"]
+        read_only_fields = ["id", "state", "has_been_created_with_order"]
 
     def create(self, validated_data):
         """
-        Retrieve the course run ressource through the provided resource_link
-        then try to create the enrollment ressource.
+        Retrieve the course run resource through the provided id
+        then try to create the enrollment resource.
         """
-        resource_link = self.initial_data["course_run"]
+        course_run_id = self.initial_data["course_run"]
```

Comment:
Actually, shouldn't we use "validated_data" here?

Reply:
"course_run" field is read_only so once data have been validated, this property is removed.

Reply:
Wait now I don't get why we need all the override in this method and not just make course_run writable (except on update which is already taken care of in the override of the update method)

Reply:
Because for the course_run field we use a `CourseRunSerializer` so the client should send the whole course run to create to enroll. So with this syntax we are able to nest course run within enrollment and in other hand create enrollment just by providing the course run id.

Reply:
ok! Maybe clearer to look into request.data.get("course_run")?
+ add explanations about this nested field context that forces you to do the create this way.

Reply:
Hmmm, from my knowledge, we do not have access to the request instance within the `create` method until we pass it explicitly through the serializer context.

Reply:
ok!